x-pack/plugins/security_solution/public/management/components/effected_policy_select/utils.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { PolicyData } from '../../../../common/endpoint/types';
+import { EffectedPolicySelection } from './effected_policy_select';
+
+export const GLOBAL_POLICY_TAG = 'policy:all';
+
+export function getArtifactTagsByEffectedPolicySelection(
+  selection: EffectedPolicySelection
+): string[] {
+  if (selection.isGlobal) {
+    return [GLOBAL_POLICY_TAG];
+  }
+  return selection.selected.map((policy) => {
+    return `policy:${policy.id}`;
+  });
+}
+
+export function getEffectedPolicySelectionByTags(
+  tags: string[],
+  policies: PolicyData[]
+): EffectedPolicySelection {
+  if (tags.length === 0 || tags.find((tag) => tag === GLOBAL_POLICY_TAG)) {
+    return {
+      isGlobal: true,
+      selected: [],
+    };
+  }
+  const selected: PolicyData[] = tags.reduce((acc, tag) => {
+    // edge case: a left over tag with a non-existed policy
+    // will be removed by veryfing the policy exists
+    const id = tag.split(':')[1];
+    const foundPolicy = policies.find((policy) => policy.id === id);
+    if (foundPolicy !== undefined) {
```

Comment:
If the policy is not found (policy or integration were removed and not unlinked from the artifact for some reason) we should ensure we remove the relation to keep the artifact clean. A part from that, would we useful to log something, maybe a `warn`? This is an edge case shouldn't happen, but you know, it could.

Reply:
perhaps the variable name is missleading but this piece of code is not actually validating that the policy exists in the system, merely that the tag follows the pattern `policy:id`. if a policy is not found is because the tag doesn't have that format.

Somehow half way what we talked about other tags could be present in the future there.